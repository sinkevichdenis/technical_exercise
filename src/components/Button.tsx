import React from 'react';
import {
  StyleProp, StyleSheet, Text, TextStyle, TouchableOpacity, ViewStyle
} from 'react-native';

import { Colors } from 'styles/colors';
import { DEFAULT_BORDER_RADIUS, DEFAULT_MARGIN } from 'styles/consts';


type Props = {
  onPress: () => void,
  text: string,
  containerStyle?: StyleProp<ViewStyle>,
  textStyle?: StyleProp<TextStyle>,
  testID?: string,
}

export const Button: React.FC<Props> = React.memo(({
  onPress,
  text,
  containerStyle,
  textStyle,
  testID = 'button',
}) => {
  return (
    <TouchableOpacity
      onPress={ onPress }
      testID={ testID }
      style={ [styles.container, containerStyle] }
    >
      <Text style={ [styles.text, textStyle] }>{ text }</Text>
    </TouchableOpacity>
  );
});

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.TEXT_BLACK,
    borderRadius: DEFAULT_BORDER_RADIUS,
    borderWidth: 1,
    paddingHorizontal: DEFAULT_MARGIN,
    paddingVertical: DEFAULT_MARGIN / 2,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: Colors.WHITE,
    fontSize: 16,
    lineHeight: 20,
  },
});
