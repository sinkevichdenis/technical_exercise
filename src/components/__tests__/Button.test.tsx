import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';

import { Button } from '../Button';


describe('BackButton component', () => {
  const mockedOnPress = jest.fn();
  const buttonText = 'test button';

  const componentRender = () => render(
    <Button
      onPress={ mockedOnPress }
      text={ buttonText }
    />
  );

  it('should match snapshot', () => {
    const { toJSON } = componentRender();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should handle press', () => {
    const { getByTestId } = componentRender();
    fireEvent.press(getByTestId('button'));
    expect(mockedOnPress).toHaveBeenCalled();
  });

  it('should show text', () => {
    const { getByText } = componentRender();
    expect(getByText(buttonText)).toBeTruthy();
  });
});
