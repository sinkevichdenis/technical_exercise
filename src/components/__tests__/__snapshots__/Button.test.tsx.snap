// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1D2330",
      "borderRadius": 12,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  testID="button"
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 16,
          "lineHeight": 20,
        },
        undefined,
      ]
    }
  >
    test button
  </Text>
</View>
`;
